utilities/release-osci.py

```diff
@@ -0,0 +1,277 @@
+        """
+        Wrapper around file / directory copying. If dry_run is set,
+        only the log message is printed and copying does not actually occur
+        """
+        print(f"[COPY ] {src} ->\n        {dest}")
+        if self.dry_run:
+            return
+
+        shutil.copy(src, dest)
+
+    def mkdir(self, path: str):
+        """
+        Wrapper around directory creation. If dry_run is set, only the log
+        message is printed and creation does not actually occur
+        """
+        print(f"[MKDIR] {path}")
+
+        if self.dry_run:
+            return
+
+        os.mkdir(path)
+
+    def update(self):
+        """
+        Runs make update from the repository root directory. If dry_run is
+        set only the log message is printed and make is not run
+        """
+        print(f"[EXEC ] make -C {self.root} update")
+
+        if self.dry_run:
+            return
+
+        subprocess.check_call(["make", "-C", self.root, "update"])
+
+    def copy_config(self):
+        """
+        Copies the master configuration as it currently stands, to a release
+        configuration based on the configured release version.
+
+        The release config is then modified in the following ways:
+
+            - image promotion is removed
+            - all pre-submit tests are removed
+            - remaining post-submit tests are modified to use versioned
+              workflows.
+        """
+        master_config = self.config_from_root("stackrox-collector-master.yaml")
+        release_config = self.config_from_root(
+            f"stackrox-collector-{self.version_str}.yaml"
+        )
+
+        self.copy(master_config, release_config)
+
+        if self.dry_run:
+            print(
+                f"[WRITE] would modify references in stackrox-collector-{self.version_str}.yaml"
+            )
+            return
+
+        print(
+            f"[WRITE] modifying references in stackrox-collector-{self.version_str}.yaml"
+        )
+
+        with open(release_config) as config_file:
+            cfg = yaml.load(config_file, Loader=yaml.Loader)
+
+            # remote image promotion
+            del cfg["promotion"]
+
+            # remove all the tests not marked as post-submit
+            cfg["tests"] = [
+                test for test in cfg["tests"] if test.get("postsubmit", False)
+            ]
+
+            for test in cfg["tests"]:
+                if "steps" not in test:
+                    continue
+
+                if "workflow" in test["steps"]:
+                    workflow = test["steps"]["workflow"]
+                    workflow = workflow.replace(
+                        "stackrox-collector-", f"stackrox-collector-{self.version_str}-"
+                    )
+                    test["steps"]["workflow"] = workflow
+
+            cfg_data = yaml.dump(cfg, Dumper=yaml.Dumper)
+
+        # dump the config back to disk
+        with open(release_config, "w+") as config_file:
```

Comment:
If we are just dumping, why do we need the `+`?

Reply:
I think just left over from testing - will remove!